packages/block-editor/src/components/inserter/block-patterns.js

```diff
@@ -73,29 +94,32 @@ function BlockPatterns( { patterns, onInsert, filterValue } ) {
 		);
 	}, [] );
 
-	return !! filteredPatterns.length ? (
+	if ( filterValue ) {
+		return !! filteredPatterns.length ? (
+			<InserterPanel title={ __( 'Search Results' ) }>
+				<BlockPatternList
+					patterns={ filteredPatterns }
+					onClickPattern={ onClickPattern }
+				/>
+			</InserterPanel>
+		) : (
+			<InserterNoResults />
+		);
+	}
+
+	return blockPatternCategories.map( ( patternCategory ) => (
 		<InserterPanel
-			title={
-				filterValue
-					? __( 'Search Results' )
-					: _x( 'All', 'patterns categories' )
-			}
+			key={ patternCategory.name }
+			title={ patternCategory.label }
```

Comment:
I think we should make the category list be a `select` and just render the first (or last accessed) category. What do you think? cc @jasmussen @pablohoneyhoney

Reply:
i fixed the performance issue related to this. So whichever direction we want, we can decide to land this as is and address the UI separately.

Reply:
Yes, agree on doing it separately.

Reply:
I think the select box with a list of categories is a fine approach to this, as we know it will scale endlessly. 👍 👍